Make operation call accept Stack& instead of Stack*

zhxchen17 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #63414

Summary:
Misuse of raw pointer in here where stack is never nullable.

Test Plan:
compiles.

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30375410](https://our.internmc.facebook.com/intern/diff/D30375410)

zhxchen17 committed:
[jit] Make operation call accept Stack& instead Stack*

Summary:
Misuse of raw pointer in here where stack is never nullable.

Test Plan:
compiles.

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

zhxchen17 commented:
@zhxchen17 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30375410).

zhxchen17 committed:
Update on "[jit] Make operation call accept Stack& instead of Stack*"


Summary:
Misuse of raw pointer in here where stack is never nullable.

Test Plan:
compiles.

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30375410](https://our.internmc.facebook.com/intern/diff/D30375410)

[ghstack-poisoned]

zhxchen17 commented:
@zhxchen17 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30375410).

SplitInfinity reviewed (approved):
Nice! Straightforward, positive change. I don't know why this was implemented with `Stack*` in the first place.

SplitInfinity reviewed (changes requested):
Requesting changes as per internal discussion

zhxchen17 committed:
Update on "[jit] Make operation call accept Stack& instead of Stack*"


Summary:
Misuse of raw pointer in here where stack is never nullable.

Test Plan:
compiles.

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30375410](https://our.internmc.facebook.com/intern/diff/D30375410)

[ghstack-poisoned]

zhxchen17 commented:
@zhxchen17 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30375410).

zhxchen17 commented on aten/src/ATen/core/stack.h:
```diff
@@ -9,7 +10,38 @@ namespace jit {
+
+class Operation {
+ public:
+  template <typename F,
```
Instead of switching over to Stack&, we can just add another constructor for Operation, such that it can take both Stack* and Stack& callback. For Stack* constructor we can make it explicit that it's deprecated, which shouldn't be a hard break for users.

Note that we wrap another level of lambda for Stack* callback to make the implementation uniform, which may have performance overhead but we can take it as a forcing factor for users to switch to Stack& callback. (or it can be inlined, I don't know).

zhxchen17 committed:
Update on "[jit] Make operation call accept Stack& instead of Stack*"


Summary:
Misuse of raw pointer in here where stack is never nullable.

Test Plan:
compiles.

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30375410](https://our.internmc.facebook.com/intern/diff/D30375410)

[ghstack-poisoned]

zhxchen17 commented:
@zhxchen17 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30375410).

zhxchen17 committed:
Update on "[jit] Make operation call accept Stack& instead of Stack*"


Summary:
Misuse of raw pointer in here where stack is never nullable.

Test Plan:
compiles.

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30375410](https://our.internmc.facebook.com/intern/diff/D30375410)

[ghstack-poisoned]

zhxchen17 commented:
@zhxchen17 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30375410).

zhxchen17 committed:
Update on "[jit] Make operation call accept Stack& instead of Stack*"


Summary:
Misuse of raw pointer in here where stack is never nullable.

Test Plan:
compiles.

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30375410](https://our.internmc.facebook.com/intern/diff/D30375410)

[ghstack-poisoned]

zhxchen17 commented:
@zhxchen17 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30375410).

zhxchen17 committed:
Update on "[jit] Make operation call accept Stack& instead of Stack*"


Summary:
Misuse of raw pointer in here where stack is never nullable.

Test Plan:
compiles.

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30375410](https://our.internmc.facebook.com/intern/diff/D30375410)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit] Make operation call accept Stack& instead of Stack*"


Summary:
Misuse of raw pointer in here where stack is never nullable.

Test Plan:
compiles.

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30375410](https://our.internmc.facebook.com/intern/diff/D30375410)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit] Make operation call accept Stack& instead of Stack*"


Summary:
Misuse of raw pointer in here where stack is never nullable.

Test Plan:
compiles.

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30375410](https://our.internmc.facebook.com/intern/diff/D30375410)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit] Make operation call accept Stack& instead of Stack*"


Summary:
Misuse of raw pointer in here where stack is never nullable.

Test Plan:
compiles.

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30375410](https://our.internmc.facebook.com/intern/diff/D30375410)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit] Make operation call accept Stack& instead of Stack*"


Summary:
Misuse of raw pointer in here where stack is never nullable.

Test Plan:
compiles.

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30375410](https://our.internmc.facebook.com/intern/diff/D30375410)

[ghstack-poisoned]
